### DISCUSSION TITLE
TSML custom address override prevented PHP version upgrade

### DISCUSSION THREAD

**Discussion by landolfi-us** [code4recovery/12-step-meeting-list#1118] [General]
We were trying to upgrade our Wordpress site to PHP 8.1 from 7.4 on Dreamhost and the conversion process was barfing on a custom address override that worked fine as a code block within our Blue Planet functions.php. This is the error the upgrade failure threw:

Fatal error: Uncaught TypeError: array_merge(): Argument #1 must be of type array, null given in
/home/dh_3fpg2i/district10nia.org/wp-content/plugins/12-step-meeting-list/includes/functions.php:227
Stack trace:
#0
/home/dh_3fpg2i/district10nia.org/wp-content/plugins/12-step-meeting-list/includes/functions.php(227):
array_merge(NULL, Array)
#1
/home/dh_3fpg2i/district10nia.org/wp-content/themes/blue-planet/functions.php(218):
tsml_custom_addresses(Array)
#2 /home/dh_3fpg2i/district10nia.org/wp-settings.php(591):
include('/home/dh_3fpg2i...')
#3 /usr/local/wp/php/WP_CLI/Runner.php(1338):
require('/home/dh_3fpg2i...')
#4 /usr/local/wp/php/WP_CLI/Runner.php(1256):
WP_CLI\Runner->load_wordpress()
#5 /usr/local/wp/php/WP_CLI/Bootstrap/LaunchRunner.php(28):
WP_CLI\Runner->start()
#6 /usr/local/wp/php/bootstrap.php(78):
WP_CLI\Bootstrap\LaunchRunner->process(Object(WP_CLI\Bootstrap\BootstrapState))
#7 /usr/local/wp/php/wp-cli.php(32): WP_CLI\bootstrap()
#8 /usr/local/wp/php/boot-fs.php(17): require_once('/usr/local/wp/p...')
#9 {main}
  thrown in
/home/dh_3fpg2i/district10nia.org/wp-content/plugins/12-step-meeting-list/includes/functions.php
on line 227
Error: There has been a critical error on this website.Learn more about troubleshooting WordPress. There has been a critical error on this website.

We were able to upgrade by temporarily removing the override and adding it back.

**Comment by kiyote33** [comment]
Hi @landolfi-us,

I'm unable to replicate your issue. I loaded up my sandbox with  PHP 8.1.9, WordPress 6.2, the Blue Planet theme, and 400+ records in TSML. Our plugin appears to work fine without any of the issues you reported. I'm wondering if there might be a conflict with other plugins you have loaded? If you could deactivate your plugins and turn them back on one at a time to see if the issue still persists it might provide a clue as to the origin of the problem.

**Comment by joshreisner** [comment]
i switched our demo site to run on PHP 8.1 and gave it the [same address override](https://wordpress.org/plugins/12-step-meeting-list/#i%20need%20to%20correct%20a%20meeting%20address%20or%20change%20a%20pin%27s%20location) from our documentation, seems to be working ok. 

could you share the override you're using? you only encountered this during the dreamhost upgrade but not once it's been upgraded? this may be hard for us to reproduce
